controllers/vsl.go

```diff
@@ -110,53 +110,37 @@ func (r *DPAReconciler) LabelVSLSecrets(log logr.Logger) (bool, error) {
 
 func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataProtectionApplication) (bool, error) {
 	for i, vslSpec := range dpa.Spec.SnapshotLocations {
+		vslYAMLPath := fmt.Sprintf("spec.snapshotLocations[%v]", i)
+		veleroVSLYAMLPath := vslYAMLPath + ".velero"
+
 		if vslSpec.Velero == nil {
 			return false, errors.New("snapshotLocation velero configuration cannot be nil")
 		}
-		vsl := velerov1.VolumeSnapshotLocation{
-			ObjectMeta: metav1.ObjectMeta{
-				// TODO: Use a hash instead of i
-				Name:      fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1),
-				Namespace: r.NamespacedName.Namespace,
-			},
-			Spec: *vslSpec.Velero,
-		}
 
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
 			vslSpec.Velero.Provider != Azure {
-			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
-
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			return false, fmt.Errorf("DPA %s.provider %s is invalid: only AWS, GCP and Azure are supported", veleroVSLYAMLPath, vslSpec.Velero.Provider)
 		}
 
 		//AWS
 		if vslSpec.Velero.Provider == AWSProvider {
 			//in AWS, region is a required field
 			if len(vslSpec.Velero.Config[AWSRegion]) == 0 {
-				return false, errors.New("region for AWS VSL is not configured, please ensure a region is configured")
+				return false, fmt.Errorf("region for AWS VSL in DPA %s.Config is not configured, please ensure a region is configured", veleroVSLYAMLPath)
```

Comment:
Typo in `config` (better to be lowercase, matching json tag, so commands like `oc get dpa velero-sample -n openshift-adp -o=jsonpath="{.spec.snapshotLocations[0].velero.config}"` work)